src/core/compact_object.h

```diff
@@ -536,6 +539,15 @@ class CompactObjectView {
   CompactObj obj_;
 };
 
+class CompactObjTypeConverter {
+ public:
+  static std::string_view ObjTypeToString(CompactObjType type);
+  static CompactObjType ObjTypeFromString(std::string_view sv);
+
+ private:
+  static const std::unordered_map<CompactObjType, std::string_view> kObjTypeToString;
+};
```

Comment:
Can be just part of the source file, no need to expose it in the header. Also I assume just having two functions like before is perferctly fine, if you don't want to expose them you can keep them in the detail / a separate namespace